lib/valueflow.cpp

```diff
@@ -1112,8 +1112,8 @@ static void valueFlowArrayBool(TokenList *tokenlist)
         if (withNonZeroOp(tok) && Token::Match(tok->astParent(), "%comp%"))
             continue;
         // TODO: Check for function argument
-        if (astIsIntegral(tok, false) || Token::Match(tok->astParent(), "&&|%oror%|%comp%") ||
-            (astIsIntegral(tok->astParent(), false) && !Token::Match(tok->astParent(), "(|%name%")) ||
+        if (astIsBool(tok) || Token::Match(tok->astParent(), "&&|%oror%|%comp%") ||
```

Comment:
The first condition `astIsBool(tok)` looks a bit suspicious. The `ValueType` should probably set the type to `bool` if there is pointer-to-bool conversion but as far as I see that does not happen currently. Did you add this for future use?

The second condition `Token::Match(...)` looks redundant. The `astIsBool(tok->astParent())` should be true as far as I see for && || %comp% operators.

Commenting out the first two conditions does not cause any test failures:

    if (/*astIsBool(tok) || Token::Match(tok->astParent(), "&&|%oror%|%comp%") ||*/